### PR TITLE
Make the FlowReference wrapper unnecessary

### PR DESCRIPTION
**Pull request by swankjesse**
This drops support for Flows as fields in objects passed
through Zipline services. But flows can be fields in any
parameter or return type. This is the same limitation of
ZiplineService instances.

All flows passed to or returned from a ZiplineService is
exactly one-shot - not consuming or double-consuming are
both unsupported.

### CODE REVIEW

**Comment by JakeWharton** [comment]
Nice. I got like 2/3rds of the way doing this a few months ago. Glad you took it across the line.

Will review tomorrow.

File: zipline/testing/src/jsMain/kotlin/app/cash/zipline/testing/sealedClassSerializersJs.kt [hunk 1]
```diff
@@ -31,14 +29,12 @@ class JsSealedClassMessageService : SealedClassMessageService {
...
-  override fun colorSwapFlow(request: FlowReference<SealedMessage>): FlowReference<SealedMessage> {
-    val requestFlow = request.take()
-    val flow: Flow<SealedMessage> = flow {
-      requestFlow.collect { message ->
+  override fun colorSwapFlow(flow: Flow<SealedMessage>): Flow<SealedMessage> {
+    return flow {
+      flow.collect { message ->
         emit(colorSwap(message))
       }
     }
```
Comment (JakeWharton) [inline zipline/testing/src/jsMain/kotlin/app/cash/zipline/testing/sealedClassSerializersJs.kt]: This is just a `flow.map(::colorSwap)` now, if you want.

**Review by JakeWharton — APPROVED** [review]
Such a big ergonomic win

File: zipline/testing/src/jsMain/kotlin/app/cash/zipline/testing/sealedClassSerializersJs.kt (hunk 1 above)
Comment (swankjesse) [reply zipline/testing/src/jsMain/kotlin/app/cash/zipline/testing/sealedClassSerializersJs.kt]: Ooooh yes please.

**Pull request merged**